Sources/MapboxCoreNavigation/NavigatorProvider.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+import MapboxNavigationNative
+import MapboxDirections
+
+/// A container class for retaining `Navigator` instance.
+///
+/// Tupically obtained via `NavigatorProvider.sharedWeakNavigator()` for storing and/or configuring shared `Navigator`.
+public final class NavigatorWithHistory {
+    let navigator: Navigator
+    let history: HistoryRecorderHandle
+    
+    init(navigator: Navigator, history: HistoryRecorderHandle) {
+        self.navigator = navigator
+        self.history = history
+    }
+}
+
+/// A fabric, used for obtaining and/or configuring `Navigator`.
+public class NavigatorProvider {
```

Comment:
Does this class need to be public? It seems to only be used inside private implementations.

Reply:
Since `Navigator` is used in `RouteController` and `PassiveLocationDataSource` and is not exposed publicly I've modified it and its properties/methods to be internal.